sway/input/cursor.c

```diff
@@ -97,6 +97,24 @@ struct sway_node *node_at_coords(
 	double ox = lx, oy = ly;
 	wlr_output_layout_output_coords(root->output_layout, wlr_output, &ox, &oy);
 
+	if (server.session_lock.locked) {
+		if (server.session_lock.lock == NULL) {
+			return NULL;
+		}
+		struct wlr_session_lock_surface_v1 *lock_surf;
+		wl_list_for_each(lock_surf, &server.session_lock.lock->surfaces, link) {
+			double _sx = ox, _sy = oy;
```

Comment:
Why do you copy those values? `wlr_surface_point_accepts_input` should not modify them.

Reply:
Right. I've switched the code to `wlr_surface_surface_at()` to account for sub-surfaces.